File: BinarytoDecimal.js
function BinaryToDecimal(binaryNumber) {
	var value = binaryNumber.toString(10);
	var length = value.length;
	var output = 0;
	for (position = 0; position < length; position++) {
		char = value.charAt(position);
		if (char == '1') {
			output += Math.pow(2, length - 1 - position);
		} else if (char == '0') {
			continue;
		} else {
			return "Make sure you only input binary values";
		}
	}
	return output;
}

const readline = require('readline');

r1 = readline.createInterface({
	input: process.stdin,
	output: process.stdout
});
r1.question("Binary Number: ", function(num) {
	console.log(BinaryToDecimal(num));
	r1.close();
});

File: BinarytoDecimal.scala
// scalac BinarytoDecimal.scala; scala BinaryToDecimal

object BinaryToDecimal {
    def BinarytoDecimal(binaryNumber: Int): Int = {
        var value = binaryNumber.toString
        var len = value.length
        var output = 0
        for (position <- 0 to len - 1) {
            var char = value(position)
            if (char == '1') {
                output += math.pow(2, len - position - 1).toInt
            } else if (char == '0') {
                output += 0
            } else {
                println("Make sure you only input binary values")
                System.exit(-1)
            }
        }
        return output
    }

    def main(args: Array[String]) = {
        print("Binary Number: ")
        val binaryNumber = scala.io.StdIn.readInt()
        println(BinarytoDecimal(binaryNumber))
    }
}